### PR TITLE
Set Up Spinning Loader Component

### PR DESCRIPTION
**Pull request by noahub**
## Description
This PR adds the SpinningLoader component to the component library

## Screencast
https://user-images.githubusercontent.com/12600902/177663364-217be100-e770-43f3-bb9d-837dbe628ce7.mov

### CODE REVIEW

File: packages/constellation/src/components/Base/Loaders/SpinningLoader/SpinningLoader.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  size = 'default',
+  variant = 'default',
+}) => {
+  return (
+    <div
+      className={cx([
+        sizeClasses[size],
+        variantClasses[variant],
+        borderWidthClasses[borderWidth],
+        'relative overflow-hidden rounded-[50%] animate-spin-fast',
```
Comment (ethanjc) [inline packages/constellation/src/components/Base/Loaders/SpinningLoader/SpinningLoader.tsx]: `rounded-full` should have the same effect as `rounded-[50%]` without adding a new className

**Review by ethanjc — DISMISSED** [review]
One tiny tailwind nitpick but otherwise looks GOOD!

**Comment by pedroapfilho** [comment]
is it possible to add a prop for the color? 🥸

**Comment by noahub** [comment]
> is it possible to add a prop for the color? 🥸

Hey @pedroapfilho, when you say color do you mean the primary and secondary border colors of the loader element? Are you looking for props that will allow you to pass arbitrary values to change these colors? Or are you looking for more variety of variants in addition to the `default` (white on white) and `minimal` (blue on blue) variants?

Personally I would be more keen on designing and adding more variants. Opening up the interface to arbitrary colors feels like it might be a little risky in allowing for undesirable color combinations. It also might add complexity in terms of how to handle light/dark mode.

But if you expect a lot of variance in the color of the loader then arbitrary colors might be the way to go.

**Comment by ethan-bc** [comment]
@noahub 
the loaders for now should just be light and dark mode, both use blues.

| -- | Light Mode | Dark Mode |
| ----------- | ----------- | ----------- |
| track | Base.Light | Darkmode.700 |
| loader | Base.Blue | Base.Blue.Muted |

I'm actually unsure how you're defining colors so i just put what we've named them in Figma. 

image

**Comment by noahub** [comment]
> @noahub the loaders for now should just be light and dark mode, both use blues.
> 
> --	Light Mode	Dark Mode
> track	Base.Light	Darkmode.700
> loader	Base.Blue	Base.Blue.Muted
> I'm actually unsure how you're defining colors so i just put what we've named them in Figma.
> 
> image

Thanks for clarifying @ethan-bc. I grabbed the colors based on the button component loaders. Is there a variant needed for the loader seen here: 
Screen Shot 2022-07-08 at 10 40 27 AM

**Review by ethanjc — APPROVED** [review]

**Pull request merged**